reverse_engineering/databaseService/databaseService.js

```diff
@@ -38,63 +33,20 @@ const getConnectionClient = async (connectionInfo, logger) => {
 	const clientId = '0dc36597-bc44-49f8-a4a7-ae5401959b85';
 	const redirectUri = 'http://localhost:8080';
 
-	switch (connectionInfo.authMethod) {
-		case 'Username / Password':
-			return sql.connect({
-				...commonConfig,
-				...credentialsConfig,
-				options: {
-					encrypt: true,
-					enableArithAbort: true,
-				},
-			});
-		case 'Username / Password (Windows)':
-			return sql.connect({
-				...commonConfig,
-				...credentialsConfig,
-				domain: connectionInfo.userDomain,
-				options: {
-					encrypt: false,
-					enableArithAbort: true,
-				},
-			});
-		case 'Azure Active Directory (MFA)':
-			const token = await getToken({ connectionInfo, tenantId, clientId, redirectUri, logger });
-			logAuthTokenInfo({ token, logger });
-			return sql.connect({
-				...commonConfig,
-				options: {
-					encrypt: true,
-					enableArithAbort: true,
-				},
-				authentication: {
-					type: 'azure-active-directory-access-token',
-					options: {
-						token,
-					},
-				},
-			});
-		case 'Azure Active Directory (Username / Password)':
-			return sql.connect({
-				...commonConfig,
-				...credentialsConfig,
-				options: {
-					encrypt: true,
-					enableArithAbort: true,
-				},
-				authentication: {
-					type: 'azure-active-directory-password',
-					options: {
-						userName: connectionInfo.userName,
-						password: connectionInfo.userPassword,
-						tenantId,
-						clientId,
-					},
-				},
-			});
-	}
+	const connection = getConnection({
+		type: connectionInfo.authMethod,
+		data: {
```

Comment:
It would be nice to have the parameters more explicit.
I would refactor it to smth like this:
```
const connection = getConnection({
    authMethod: connectionInfo.authMethod,
    connectionInfo,
    commonConfig,
    credentialsConfig,
    tenantId,
    clientId,
    redirectUri,
    logger,
});
```